### PR TITLE
feat: 95351 - [Telemetry] Add debug log

### PR DESCRIPTION
**Pull request by dvatan-plenty**
feat: AB#95351: [Telemetry] Add debug log

### CODE REVIEW

**Review by pduca-plenty — DISMISSED** [review]

File: pkg/teldrvr/newrelicAPM.go [hunk 1]
```diff
@@ -71,12 +71,13 @@ type NewRelicSegmentContainer struct {
...
-	trace            string
-	traceID          string
-	processID        string
+	transaction            *newrelic.Transaction
+	segmentContainer       NewRelicSegmentContainer
+	attributes             map[string]any
+	trace                  string
+	traceID                string
+	processID              string
+	segmentsStartWasLogged map[string]struct{}
```
Comment (mbrueschke) [inline pkg/teldrvr/newrelicAPM.go]: why was this added?

File: pkg/teldrvr/newrelicAPM.go [hunk 2]
```diff
@@ -194,6 +195,12 @@ func (t *APMTransaction) Info(_ string, readCloser io.ReadCloser) error {
 	return nil
 }
 
+// Debug [NOT IMPLEMENTED]
+func (t *APMTransaction) Debug(_ string, readCloser io.ReadCloser) error {
+	// TODO - Will be implemented as in Error()
```
Comment (mbrueschke) [inline pkg/teldrvr/newrelicAPM.go]: why will this be implemented as in Error()?

Have you taken a look into t.transaction.RecordLog() for logging

Comment (mbrueschke) [reply pkg/teldrvr/newrelicAPM.go]: if possible we might want to be able to log debug and info with the record log

**Review by mbrueschke — CHANGES_REQUESTED** [review]

File: pkg/teldrvr/newrelicAPM.go [hunk 3]
```diff
@@ -191,6 +192,51 @@ func (t *APMTransaction) Error(_ string, readCloser io.ReadCloser) error {
...
+		}
+	}()
+
+	bytesRead, err := readCloser.Read(InfoMsg)
+	if err != nil {
+		return errors.New("error while reading Debug message")
+	}
+
+	recordLog := newrelic.LogData{}
+	recordLog.Severity = "Debug"
```
Comment (mbrueschke) [inline pkg/teldrvr/newrelicAPM.go]: info

File: pkg/teldrvr/newrelicAPM.go [hunk 4]
```diff
@@ -191,6 +191,51 @@ func (t *APMTransaction) Error(_ string, readCloser io.ReadCloser) error {
...
+	recordLog := newrelic.LogData{
+		Severity: "Info",
+		Message:  string(infoMsg[:bytesRead]),
+	}
+
+	t.transaction.RecordLog(recordLog)
+	return nil
+}
+
+// Debug [NOT IMPLEMENTED]
```
Comment (mbrueschke) [inline pkg/teldrvr/newrelicAPM.go]: comment says that this function is not implemented ^^

**Review by mbrueschke — DISMISSED** [review]

**Review by mbrueschke — APPROVED** [review]

**Pull request merged**